### PR TITLE
Update originalBlock session key once for each block in BatchFileTransformerPipe

### LINKED ISSUE #3260: SessionKeys are not unique/ overwritten when content is changed in ladybug
**Describe the issue**
Too many SessionKeys in the ladybug

**Reporter**
Niels

**To Reproduce**
Start IAF-test
Run Larva scenario 'BatchFileTransformerPipe'
Look at the ladybug

**Screenshots**
image

### PR DESCRIPTION
**Pull request by alisihab**
Closes #3260

### CODE REVIEW

File: core/src/main/java/nl/nn/adapterframework/batch/IResultHandler.java [hunk 1]
```diff
@@ -72,8 +73,8 @@ public interface IResultHandler extends IConfigurable {
 	 */
 	void closeRecordType(PipeLineSession session, String streamId) throws Exception;
 	
-	void openBlock(PipeLineSession session, String streamId, String blockName) throws Exception;
-	void closeBlock(PipeLineSession session, String streamId, String blockName) throws Exception;
+	void openBlock(PipeLineSession session, String streamId, String blockName, Map<String, Object> blocks) throws Exception;
+	void closeBlock(PipeLineSession session, String streamId, String blockName, Map<String, Object> blocks) throws Exception;
```
Comment (gvanbrakel) [inline core/src/main/java/nl/nn/adapterframework/batch/IResultHandler.java:77]: Is the session parameter really used somewhere? Can it be removed?

**Review by gvanbrakel — CHANGES_REQUESTED** [review]

**Review by gvanbrakel — APPROVED** [review]

File: core/src/main/java/nl/nn/adapterframework/batch/BatchFileTransformerPipe.java [hunk 2]
```diff
@@ -81,32 +88,36 @@ public PipeRunResult doPipe(Message input, PipeLineSession session) throws PipeR
...
+
 		if (!(input.asObject() instanceof String)) {
 			throw new PipeRunException(this,"expected String containing filename as input, got ["+ClassUtils.nameOf(input)+"], value ["+input+"]");
 		}
-		String filename	= (String)input.asObject();
+		String filename;
+		try {
+			filename = input.asString();
+		} catch (IOException e) {
+			throw new PipeRunException(this, "could not get message ["+input+"] as String", e);
```
Comment (nielsm5) [inline core/src/main/java/nl/nn/adapterframework/batch/BatchFileTransformerPipe.java:99]:
```suggestion
			throw new PipeRunException(this, "could not read message ["+input+"] as String", e);
```

File: core/src/main/java/nl/nn/adapterframework/batch/BatchFileTransformerPipe.java [hunk 3]
```diff
@@ -52,25 +53,31 @@ public class BatchFileTransformerPipe extends StreamTransformerPipe {
...
 	protected InputStream getInputStream(String streamId, Message input, PipeLineSession session) throws PipeRunException {
 		try {
-			String filename	= (String)input.asObject();
+			String filename = input.asString();
 			File file = new File(filename);
 			return new FileInputStream(file);
 		} catch (FileNotFoundException e) {
 			throw new PipeRunException(this,"cannot find file ["+streamId+"]",e);
+		} catch (IOException e) {
+			throw new PipeRunException(this, "could not get message ["+input+"] as String", e);
```
Comment (nielsm5) [inline core/src/main/java/nl/nn/adapterframework/batch/BatchFileTransformerPipe.java:75]:
```suggestion
			throw new PipeRunException(this, "could not read message ["+input+"] as String", e);
```

File: core/src/main/java/nl/nn/adapterframework/batch/BatchFileTransformerPipe.java [hunk 4]
```diff
@@ -52,25 +53,31 @@ public class BatchFileTransformerPipe extends StreamTransformerPipe {
 
 	@Override
 	protected String getStreamId(Message input, PipeLineSession session) {
-		String filename = (String)input.asObject();
+		String filename = null;
+		try {
+			filename = input.asString();
+		} catch (IOException e) {
+			log.error("Could not get message ["+input+"] as String");
```
Comment (nielsm5) [inline core/src/main/java/nl/nn/adapterframework/batch/BatchFileTransformerPipe.java:60]:
```suggestion
			log.error("Could not read message ["+input+"] as String");
```

**Review by nielsm5 — APPROVED** [review]

**Pull request merged**